### PR TITLE
Optimizing scheduling of the AcquireTask completion callbacks

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
Moving AcquireTask completion off the channel pool executor's event loop when pending acquisition timeouts fire to avoid spamming the channel pool executor's event loop.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java [hunk 1]
```diff
@@ -860,26 +861,52 @@ private void ensureInEventLoop() {
      *
      * @return {@code true} if the given {@link Channel channel} is serviceable; {@code false} otherwise.
      */
-    private boolean isChannelServiceable(final Channel channel, boolean ignoreMaxRequestLimit) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:863]: why don't we need `ignoreMaxRequestLimit` anymore?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java [hunk 2]
```diff
@@ -548,10 +546,10 @@ boolean hasRntbdContext() {
         return this.contextFuture.getNow(null) != null;
     }
 
-    boolean isServiceable(final int demand, boolean ignoreMaxRequestPerChannel) {
+    boolean isServiceable(final int demand) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:549]: why don't we need `ignoreMaxRequestPerChannel` anymore?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:863]: Annie borught up a good point that in the fallback-else-path we were incorrectly ignoring maxPendingRequestCount - when reeenforicng it all places passed in false anywhere (except in the high load factor part - where we pick the available channel with the least number of requests (but also should only do it if < maxPendingRequests)

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:863]: Bottom-line - no place should be ignoring maxPendingRequests

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:549]: Discussed offline - not needed anymore.

**Review by moderakh — APPROVED** [review]
thank you LGTM

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java (hunk 1 above)
Comment (kirankumarkolli) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java:863]: Closed offline

**Pull request merged**